TARS/src/mcu_main/sensors/HighGSensor.h

```diff
@@ -5,22 +5,34 @@
 #include "ChRt.h"
 #include "SparkFun_Qwiic_KX13X.h"
 
-struct Acceleration {
-    float ax, ay, az;
-};
+#include "common/packet.h"
+#include "mcu_main/error.h"
+#include "mcu_main/hilsim/HILSIMPacket.h"
+
+/**
+* 
+* @class HighGSensor
+* 
+* @brief This class initializes and controls the HighG sensor. One can obtain data using the functions provided in the class.
+* 
+* Currently the chip select is given to the default constructor using the
+* QuiicKX132. Using this class one can obtain the current acceleration. 
+* The range on the high-g sensor is better than the low-g sensor, with a range higher than could reasonably be obtained.
+*/
 
 struct HighGSensor {
    public:
     HighGSensor() = default;
 
     MUTEX_DECL(mutex);
 
-    void init();
+    ErrorCode __attribute__((warn_unused_result)) init();
     void update();
+    void update(HILSIMPacket hilsim_packet);
     Acceleration getAccel();
 
    private:
     float ax = 0.0, ay = 0.0, az = 0.0;
     systime_t timestamp = 0;
-    QwiicKX132 KX{};
+    QwiicKX134 KX;
```

Comment:
This needs to be wrapped in `#ENABLE_HIGH_G` guards, as does the include above. See comment on `GasSensor.h`